core/app/models/spree/tax_rate.rb

```diff
@@ -78,99 +47,61 @@ def self.store_pre_tax_amount(item, rates)
-    #   21% inclusive - "Clothing" - Spain
-    #   18% inclusive - "Clothing" - France
-    #   10% inclusive - "Food" - Spain
-    #   8% inclusive - "Food" - France
-    #   5% inclusive - "Hotels" - Spain
-    #   2% inclusive - "Hotels" - France
-    #
-    # Order has:
-    #   Line Item #1 - Tax Category: Clothing
-    #   Line Item #2 - Tax Category: Food
-    #
-    # Tax rates that should be selected:
-    #
-    #  21% inclusive - "Clothing" - Spain
-    #  10% inclusive - "Food" - Spain
-    #
-    # If the order's address changes to one in France, then the tax will be recalculated:
-    #
-    #  18% inclusive - "Clothing" - France
-    #  8% inclusive - "Food" - France
-    #
-    # Note here that the "Hotels" tax rates will not be used at all.
-    # This is because there are no items which have the tax category of "Hotels".
-    #
-    # Under no circumstances should negative adjustments be applied for the Spanish tax rates.
-    #
-    # Those rates should never come into play at all and only the French rates should apply.
-    def potentially_applicable?(order_tax_zone)
-      # If the rate's zone matches the order's tax zone, then it's applicable.
-      self.zone == order_tax_zone ||
-      # If the rate's zone *contains* the order's tax zone, then it's applicable.
-      self.zone.contains?(order_tax_zone) ||
-      # 1) The rate's zone is the default zone, then it's always applicable.
-      (self.included_in_price? && self.zone.default_tax)
+    def self.included_tax_amount_for(zone, category)
+      return 0 unless zone && category
+      potential_rates_for_zone(zone).
+        included_in_price.
+        for_tax_category(category).
+        pluck(:amount).sum
     end
 
     def adjust(order, item)
-      included = included_in_price && default_zone_or_zone_match?(order)
-      create_adjustment(order, item, included)
+      create_adjustment(order, item, included_in_price)
     end
 
     def compute_amount(item)
-      refund = included_in_price && !default_zone_or_zone_match?(item.order)
-      compute(item) * (refund ? -1 : 1)
+      compute(item)
     end
 
     private
 
-    def default_zone_or_zone_match?(order)
-      Zone.default_tax.try(:contains?, order.tax_zone) || order.tax_zone == zone
+    def label
+      Spree.t included_in_price? ? :including_tax : :excluding_tax,
+              scope: 'adjustment_labels.tax_rates',
+              name: name.presence || tax_category.name,
+              amount: amount_for_label
     end
 
-    def label(adjustment_amount)
-      label = ""
-      label << Spree.t(:refund) << ' ' if adjustment_amount < 0
-      label << (name.present? ? name : tax_category.name) + " "
-      label << (show_rate_in_label? ? "#{amount * 100}%" : "")
-      label << " (#{Spree.t(:included_in_price)})" if included_in_price?
-      label
+    def amount_for_label
+      return "" unless show_rate_in_label?
+      ' ' + ActiveSupport::NumberHelper::NumberToPercentageConverter.convert(
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.